src/Autofac/Core/Resolving/Pipeline/ResolveRequestContextBase.cs

```diff
@@ -59,7 +59,7 @@ public abstract class ResolveRequestContextBase : IComponentContext
             Parameters = request.Parameters;
             PhaseReached = PipelinePhase.ResolveRequestStart;
             Tracer = tracer;
-            TracingEnabled = tracer is object;
```

Comment:
@alistairjevans I remember that you've introduced this here for some performance reason. Is there a specific thing that I might miss, on why you have used `is object` instead of `tracer != null`?

Reply:
Tracer is changing a lot in my current work, switching to `System.Diagnostics.DiagnosticSource`. I wouldn't worry about this too much.

Reply:
(But I don't think there's a difference; `is object` is readability syntactic sugar, I believe.)

Reply:
Okay cool.

Reply:
Storing the bool `TracingEnabled` is an optimisation to prevent checking it repeatedly. 

The `is object` is a test for non-null that does not look at custom operators. I generally prefer it now that pattern matching performance is optimised for that case.

https://stackoverflow.com/questions/58505313/proper-null-check-on-service-result-in-c-sharp-net/58506420#58506420

Reply:
So basically `!(x is null)` would generate less IL code as well? Anyway, adding it back!

Reply:
Apparently this is also possible now
`TracingEnabled = tracer is { };`

:man_shrugging:

Reply:
Yeah, pattern matching is kinda crazy.

I think the `is object` and `is null` terms have specific JIT optimisation to make them as fast as the original null comparison. Not sure about the `is {}`.